store/data/actions.js

```diff
@@ -2,36 +2,33 @@
  * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
  * See LICENSE for license details.
  */
+import { types } from './mutations';
+
 export default {
     getDictionaries({ commit, rootState }) {
         const { language: userLanguageCode } = rootState.authentication.user;
-        const callPaths = [
-            { mutation: 'languages', path: `${userLanguageCode}/dictionary/languages` },
-            { mutation: 'currencies', path: `${userLanguageCode}/dictionary/currencies` },
-            { mutation: 'units', path: `${userLanguageCode}/dictionary/units` },
-            { mutation: 'attrTypes', path: `${userLanguageCode}/dictionary/attributes/types` },
-            { mutation: 'attrGroups', path: `${userLanguageCode}/dictionary/attributes/groups` },
-            { mutation: 'dateFormats', path: `${userLanguageCode}/dictionary/date_format` },
-            { mutation: 'imageFormats', path: `${userLanguageCode}/dictionary/image_format` },
+
+        const requestConfigs = [
+            { stateProp: 'languages', requestPath: `${userLanguageCode}/dictionary/languages` },
+            { stateProp: 'currencies', requestPath: `${userLanguageCode}/dictionary/currencies` },
+            { stateProp: 'units', requestPath: `${userLanguageCode}/dictionary/units` },
+            { stateProp: 'attrTypes', requestPath: `${userLanguageCode}/dictionary/attributes/types` },
+            { stateProp: 'attrGroups', requestPath: `${userLanguageCode}/dictionary/attributes/groups` },
+            { stateProp: 'dateFormats', requestPath: `${userLanguageCode}/dictionary/date_format` },
+            { stateProp: 'imageFormats', requestPath: `${userLanguageCode}/dictionary/image_format` },
         ];
-        const promises = [];
 
-        callPaths.forEach(({ mutation, path }) => {
-            promises.push(this.app.$axios.$get(path).then(response => new Promise(((resolve) => {
-                resolve([response, mutation]);
-            }))));
+        const promises = requestConfigs.map(({ stateProp, requestPath }) => {
```

Comment:
Maybe something like this: 
`
const promises = requestConfigs.map(({ stateProp, requestPath }) =>   this.app.$axios.$get(requestPath).then((response) => {
    commit(types.SET_CUSTOM_STATE_PROPERTY, { stateProp, value: response });
}));
`
it's a little shorter.

Reply:
Yes, you are right, but ESlint is highlighting it because of extended line length limit (100). I also think that nested arrow functions with promises and callbacks are less readable, but finally it's your decision

Reply:
yes, I think my solution might be too complicated and unclear.